pkg/certificate/signer.go

```diff
@@ -130,14 +174,339 @@ func (s *signerImpl) Stop(namespace string) {
+		logger.Info("CA is expiring soon — rotating it", "namespace", namespace, "currentVersion", currentVersion)
+		newVersion := s.incrementVersion(currentVersion)
+		logger.Info("Incrementing CA version for rotation", "namespace", namespace, "newVersion", newVersion)
+
+		return true, newVersion, nil
+	}
+
+	logger.V(log.TRACE).Info("CA is still valid — no rotation needed", "namespace", namespace, "version", currentVersion)
+
+	return false, currentVersion, nil
+}
+
+// incrementVersion increments the CA version string, handling both numeric and non-numeric versions.
+func (s *signerImpl) incrementVersion(currentVersion string) string {
+	if version, err := strconv.Atoi(currentVersion); err == nil {
+		return strconv.Itoa(version + 1)
+	}
+	// Fallback if current version is not a number
+	return "2"
+}
+
+// startPeriodicCACheck starts a periodic check for CA certificate expiration.
+func (s *signerImpl) startPeriodicCACheck(namespace string) {
+	s.caStopCh = make(chan struct{})
+
+	go wait.Until(func() {
+		logger.V(1).Info("Performing periodic CA check", "namespace", namespace)
+
+		// Check if CA secret exists first
+		caSecretClient := s.dynClient.Resource(corev1.SchemeGroupVersion.WithResource("secrets")).Namespace(namespace)
+		_, err := caSecretClient.Get(context.TODO(), CASecretName, metav1.GetOptions{})
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				logger.V(1).Info("CA secret not found during periodic check, skipping", "namespace", namespace)
+				return
+			}
+
+			logger.Error(err, "Failed to check CA secret during periodic check", "namespace", namespace)
+
+			return
+		}
+
+		// CA exists, check if it needs rotation
+		if err := s.issueCA(context.TODO(), namespace); err != nil {
+			logger.Error(err, "Failed periodic CA check", "namespace", namespace)
+		}
+	}, CACheckInterval, s.caStopCh)
+
+	logger.Info("Started periodic CA check", "interval", CACheckInterval, "namespace", namespace)
+}
+
+// resignAllCSRs finds all existing CSR secrets and re-signs them with the new CA.
+func (s *signerImpl) resignAllCSRs(ctx context.Context, namespace string) error {
+	secretClient := s.dynClient.Resource(corev1.SchemeGroupVersion.WithResource("secrets")).Namespace(namespace)
+
+	// List all secrets with the CSR request label
+	secretList, err := secretClient.List(ctx, metav1.ListOptions{
+		LabelSelector: SigningRequestLabelKey,
+	})
+	if err != nil {
+		return fmt.Errorf("failed to list CSR secrets: %w", err)
+	}
+
+	resignCount := 0
+
+	for _, item := range secretList.Items {
+		_ = util.Update(ctx, resource.ForDynamic(secretClient), &item,
+			func(existing *unstructured.Unstructured) (*unstructured.Unstructured, error) {
+				annotations := existing.GetAnnotations()
+				delete(annotations, RequestSignedLabelKey)
+				existing.SetAnnotations(annotations)
+
+				return existing, nil
+			})
+
+		resignCount++
+
+		logger.Info("Marked CSR secret for re-signing", "secret", item.GetName(), "namespace", namespace)
```

Comment:
```suggestion
		logger.Infof("Marked CSR secret \"%s/%s\" for re-signing", namespace, item.GetName())
```
PS: you're using a mix of logging styles, ie either in-line formatted or with separate key/values. Perhaps we should use one for consistency (I'm partial to in-line formatted).

Reply:
changed to in-line